### PR TITLE
PLAT-136880: Update docs

### PR DESCRIPTION
**Pull request by MikyungKim**
Enact-DCO-1.0-Signed-off-by: Mikyung Kim (mikyung27.kim@lge.com)

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Update docs prior to major version release.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
PLAT-136880

### CODE REVIEW

File: docs/testing-components/unit-testing/index.md [hunk 1]
```diff
@@ -302,7 +302,7 @@ Unfortunately, we can't always write pure functions. Sometimes we may have to us
 to make an AJAX call, etc. These are good reasons to not use pure functions. These functions are tougher to test but, not
 impossible. For example, if you are using Redux for complex application/component state management, you may be using side
 effects to get some of your data.  The Async Action Creators section in
-[http://redux.js.org/docs/recipes/WritingTests.html](http://redux.js.org/docs/recipes/WritingTests.html) shows how you can
+[http://redux.js.org/docs/recipes/WritingTests.html](https://redux.js.org/recipes/writing-tests) shows how you can
```
Comment (jeonghee27) [inline docs/testing-components/unit-testing/index.md:305]:
```suggestion
[https://redux.js.org/recipes/writing-tests](https://redux.js.org/recipes/writing-tests) shows how you can
```

Why did you fix only text?

File: docs/accessibility.md [hunk 2]
```diff
@@ -61,9 +60,9 @@ const App = () => {
 
 ```js
 import AnnounceDecorator from '@enact/ui/AnnounceDecorator';
-import React from 'react';
+import {Component} from 'react';
```
Comment (jeonghee27) [inline docs/accessibility.md:63]: Shouldn't we fix the line 45 and moonstone?

Comment (MikyungKim) [reply docs/accessibility.md:63]: Good catch for line 45. It must be undone.
The rest of the stuff about moonstone will be taken care of in PLAT-139104.

File: docs/testing-components/unit-testing/index.md (hunk 1 above)
Comment (MikyungKim) [reply docs/testing-components/unit-testing/index.md:305]: 👍

File: docs/accessibility.md (hunk 2 above)
Comment (jeonghee27) [reply docs/accessibility.md:63]: I can found another 7 `import React from 'react';` in this branch

File: docs/migration/enact/migrating-to-enact-4.md [hunk 3]
```diff
@@ -10,13 +10,61 @@ this guide.
...
+their code does not rely on features that are no longer available in these versions.
+React 17 introduced the new JSX Transform. The big change is, with the new JSX Transform,
+you can use JSX without importing React. Enact CLI 4 supports this feature on related commands
+and you will now see below lint warning after packing or linting if you didn't change your code.
+
+```js
+warning  'React' is defined but never used  no-unused-vars
+```
+
+Well, you may see lots of them! But don't be afraid. Fortunately, React provides a module that automatically converts the code on behalf of you. It's "react-codemod". Please see more details on [here](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports.).
```
Comment (jeonghee27) [inline docs/migration/enact/migrating-to-enact-4.md:24]: please remove `.` on the link
(https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports.).

Comment (MikyungKim) [reply docs/migration/enact/migrating-to-enact-4.md:24]: 👍

File: docs/migration/enact/migrating-to-enact-4.md [hunk 4]
```diff
@@ -10,13 +10,61 @@ this guide.
...
 
 ## ui
+
+### `A11yDecorator`
+The `ui/A11yDecorator` has been removed.
+
+### `Button`, `Icon`, `IconButton`, and `LabeledIcon`
+The default `size` value has been removed.
+
+### `BodyText`, `Button`, `Goup`, `Heading`, `Icon`, `IconButton`, `Image`, `ImageItem`, `LabeledIcon`, `Layout`, `ProgressBar`, `Repeater`, `Slider`, `SlotItem`, `Spinner`, `ToggleIcon`, `ToggleItem`, and `ViewManager`
```
Comment (jeonghee27) [inline docs/migration/enact/migrating-to-enact-4.md:48]: Goup

File: docs/migration/enact/migrating-to-enact-4.md [hunk 5]
```diff
@@ -10,13 +10,61 @@ this guide.
...
+The forwarding `ref`s to the repective root component support is added.
```
Comment (jeonghee27) [inline docs/migration/enact/migrating-to-enact-4.md:49]: repective

File: docs/migration/enact/migrating-to-enact-4.md (hunk 4 above)
Comment (MikyungKim) [reply docs/migration/enact/migrating-to-enact-4.md:48]: 👏

**Review by jeonghee27 — APPROVED** [review]
LGTM

File: docs/migration/enact/migrating-to-enact-4.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+their code does not rely on features that are no longer available in these versions.
+React 17 introduced the new JSX Transform. The big change is, with the new JSX Transform,
+you can use JSX without importing React. Enact CLI 4 supports this feature on related commands
+and you will now see below lint warning after packing or linting if you didn't change your code.
+
+```js
+warning  'React' is defined but never used  no-unused-vars
+```
+
+Well, you may see lots of them! But don't be afraid. Fortunately, React provides a module that automatically converts the code on behalf of you. It's `react-codemod`. Please see more details on [here](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports).
```
Comment (seunghoh) [inline docs/migration/enact/migrating-to-enact-4.md:24]: Please see more details <s>on</s> here

File: docs/migration/enact/migrating-to-enact-4.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+### `Button`, `Icon`, `IconButton`, and `LabeledIcon`
+The default `size` value has been removed.
+
+### `BodyText`, `Button`, `Group`, `Heading`, `Icon`, `IconButton`, `Image`, `ImageItem`, `LabeledIcon`, `Layout`, `ProgressBar`, `Repeater`, `Slider`, `SlotItem`, `Spinner`, `ToggleIcon`, `ToggleItem`, and `ViewManager`
+The forwarding `ref`s to the respective root component support is added.
+
+### `Touchable`
+The `onHold` and `onHoldPulse` changed to `onHoldStart` and `onHold` respectively.
+
```
Comment (seunghoh) [inline docs/migration/enact/migrating-to-enact-4.md:53]: Can we add the changing table like
| before | after |
|---|---|
| onHold | onHoldStart |
| onHoldStart | onHold |
| | onHoldEnd|

**Review by seunghoh — CHANGES_REQUESTED** [review]

File: docs/migration/enact/migrating-to-enact-4.md (hunk 7 above)
Comment (MikyungKim) [reply docs/migration/enact/migrating-to-enact-4.md:53]: 🆗

File: docs/migration/enact/migrating-to-enact-4.md (hunk 6 above)
Comment (MikyungKim) [reply docs/migration/enact/migrating-to-enact-4.md]: 👍

**Review by seunghoh — APPROVED** [review]

**Pull request merged**